deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/transferlearning/FineTuneConfiguration.java

```diff
@@ -0,0 +1,278 @@
+package org.deeplearning4j.nn.transferlearning;
+
+import lombok.Builder;
+import org.deeplearning4j.nn.api.OptimizationAlgorithm;
+import org.deeplearning4j.nn.conf.*;
+import org.deeplearning4j.nn.conf.distribution.Distribution;
+import org.deeplearning4j.nn.conf.layers.ConvolutionLayer;
+import org.deeplearning4j.nn.conf.layers.Layer;
+import org.deeplearning4j.nn.conf.layers.LayerValidation;
+import org.deeplearning4j.nn.conf.layers.SubsamplingLayer;
+import org.deeplearning4j.nn.conf.stepfunctions.StepFunction;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.activations.IActivation;
+
+import java.util.Map;
+
+/**
+ * Created by Alex on 21/02/2017.
+ */
+@Builder(builderClassName = "Builder")
+public class FineTuneConfiguration {
+
+    protected IActivation activationFn;
+    protected WeightInit weightInit;
+    protected Double biasInit;
+    protected Distribution dist;
+    protected Double learningRate;
+    protected Double biasLearningRate;
+    protected Map<Integer, Double> learningRateSchedule;
+    protected Double lrScoreBasedDecay;
+    protected Double l1;
+    protected Double l2;
+    protected Double l1Bias;
+    protected Double l2Bias;
+    protected Double dropOut;
+    protected Updater updater;
+    protected Double momentum;
+    protected Map<Integer, Double> momentumSchedule;
+    protected Double epsilon;
+    protected Double rho;
+    protected Double rmsDecay;
+    protected Double adamMeanDecay;
+    protected Double adamVarDecay;
+    protected Boolean miniBatch;
+    protected Integer numIterations;
+    protected Integer maxNumLineSearchIterations;
+    protected Long seed;
+    protected Boolean useRegularization;
+    protected OptimizationAlgorithm optimizationAlgo;
+    protected StepFunction stepFunction;
+    protected Boolean useDropConnect;
+    protected Boolean minimize;
+    protected GradientNormalization gradientNormalization;
+    protected Double gradientNormalizationThreshold;
+    protected LearningRatePolicy learningRatePolicy;
+    protected Double lrPolicyDecayRate;
+    protected Double lrPolicySteps;
+    protected Double lrPolicyPower;
+    protected ConvolutionMode convolutionMode;
+
+    protected Boolean pretrain;
```

Comment:
We want to check for nulls/nans as unset which is why it's not initialized here. Part of the logic of how to be able to override the configurations